pkg/server/controllers.go

```diff
@@ -952,10 +954,6 @@ func (s *Server) installAPIExportEndpointSliceController(ctx context.Context, co
 			logger.Error(err, "failed to finish post-start-hook")
 			return nil // don't klog.Fatal. This only happens when context is cancelled.
 		}
-		if err := s.waitForOptionalSync(hookContext.StopCh); err != nil {
-			logger.Error(err, "failed to finish post-start-hook")
-			return nil // don't klog.Fatal. This only happens when context is cancelled.
-		}
```

Comment:
What guarantees that shard informers for the cache server have synced before APIExportEndpointSliceController is started?

Reply:
does it matter?

Reply:
ack